edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/TaskStatusRequestHandler.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Requests
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Storage;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class TaskStatusRequestHandler : RequestHandlerBase<TaskStatusRequest, TaskStatusResponse>
+    {
+        static readonly Version ExpectedSchemaVersion = new Version("1.0");
+
+        public override string RequestName => "GetTaskStatus";
+
+        protected override Task<Option<TaskStatusResponse>> HandleRequestInternal(Option<TaskStatusRequest> payloadOption, CancellationToken cancellationToken)
+        {
+            TaskStatusRequest payload = payloadOption.Expect(() => new ArgumentException("Request payload not found"));
```

Comment:
Do I understand correctly that this will throw if someone calls a `GetTaskStatus` direct method on edge agent without a payload? It seems like that would be an easy way to crash edge agent...

Reply:
The error should get caught and handled by the upper layer (RequestManager).